modules/percolator/src/test/java/org/elasticsearch/percolator/PercolatorQuerySearchTests.java

```diff
@@ -221,4 +226,49 @@ public void testMapUnmappedFieldAsText() throws IOException {
         assertSearchHits(response, "1");
     }
 
+    public void testRangeQueriesWithNow() throws Exception {
+        assertAcked(client().admin().indices().prepareCreate("test")
+            .addMapping("_doc", "field1", "type=keyword", "field2", "type=date", "query", "type=percolator")
+        );
+
+        client().prepareIndex("test", "_doc", "1")
+            .setSource(jsonBuilder().startObject().field("query", rangeQuery("field2").from("now-3s").to("now+3s")).endObject())
+            .get();
+        client().prepareIndex("test", "_doc", "2")
+            .setSource(jsonBuilder().startObject().field("query", boolQuery()
+                .filter(termQuery("field1", "value"))
+                .filter(rangeQuery("field2").from("now-3s").to("now+3s"))
+            ).endObject())
+            .get();
+
+
+        Script script = new Script(ScriptType.INLINE, MockScriptPlugin.NAME, "1==1", Collections.emptyMap());
+        client().prepareIndex("test", "_doc", "3")
+            .setSource(jsonBuilder().startObject().field("query", boolQuery()
+                .filter(scriptQuery(script))
+                .filter(rangeQuery("field2").from("now-3s").to("now+3s"))
+            ).endObject())
+            .get();
+        client().admin().indices().prepareRefresh().get();
+
+        BytesReference source = BytesReference.bytes(jsonBuilder().startObject()
+            .field("field1", "value")
+            .field("field2", System.currentTimeMillis())
+            .endObject());
+        SearchResponse response = client().prepareSearch()
+            .setQuery(new PercolateQueryBuilder("query", source, XContentType.JSON))
+            .get();
+        assertHitCount(response, 3);
+
+        Thread.sleep(5000);
```

Comment:
is there another way to do this instead of using a 5 second sleep? Maybe you can play with System.currentTimeMillis()?

Reply:
Yes, I think so. I would never push this test as is to master. I created this the quickly validate whether this fix was working.